lib/devices/get_services.ex

```diff
@@ -11,6 +11,18 @@ defmodule Onvif.Devices.GetServices do
   end
 
   def response(xml_response_body) do
-    xml_response_body
+    result =
+      xml_response_body
+      |> parse(namespace_conformant: true, quiet: true)
+      |> xpath(
+        ~x"//s:Envelope/s:Body/tds:GetServicesResponse/tds:Service"el
+        |> add_namespace("s", "http://www.w3.org/2003/05/soap-envelope")
+        |> add_namespace("tds", "http://www.onvif.org/ver10/device/wsdl")
+        |> add_namespace("tt", "http://www.onvif.org/ver10/schema")
+      )
+      |> Enum.map(&Onvif.Device.Service.parse/1)
+      |> Enum.map(&Onvif.Device.Service.to_struct/1)
```

Comment:
I know I introduced the pattern of returning a keyword list with :ok and :error responses. I'm not sure I like it though. feel like we should maybe filter out errors and log them and return only the valid responses.

Reply:
Changed